openedx/features/course_search/views/course_search.py

```diff
@@ -0,0 +1,66 @@
+"""
+Views for the course search page.
+"""
+
+from django.contrib.auth.decorators import login_required
+from django.core.context_processors import csrf
+from django.core.urlresolvers import reverse
+from django.template.loader import render_to_string
+from django.utils.decorators import method_decorator
+from django.views.decorators.cache import cache_control
+from django.views.decorators.csrf import ensure_csrf_cookie
+
+from courseware.courses import get_course_overview_with_access
+from lms.djangoapps.courseware.views.views import CourseTabView
+from opaque_keys.edx.keys import CourseKey
+from openedx.core.djangoapps.plugin_api.views import EdxFragmentView
+from openedx.features.course_experience import default_course_url_name
+from util.views import ensure_valid_course_key
+from web_fragments.fragment import Fragment
+
+
+class CourseSearchView(CourseTabView):
+    """
+    The home page for a course.
+    """
+    @method_decorator(login_required)
+    @method_decorator(ensure_csrf_cookie)
```

Comment:
I have to admit to not knowing when it is needed. :-) Why would it not be needed here?